internal/biz/usecase/resources/resource_service.go

```diff
@@ -123,26 +123,25 @@ func (uc *Usecase) ReportResource(ctx context.Context, request *v1beta2.ReportRe
 	err = uc.resourceRepository.GetTransactionManager().HandleSerializableTransaction(
 		uc.resourceRepository.GetDB(),
 		func(tx *gorm.DB) error {
+			// Check for duplicate transaction ID's before we find the resource for quicker returns if it fails
+			transactionId := request.GetRepresentations().GetMetadata().GetTransactionId()
+			if transactionId != "" {
+				alreadyProcessed, err := uc.resourceRepository.HasTransactionIdBeenProcessed(tx, transactionId)
+				if err != nil {
+					return fmt.Errorf("failed to check transaction ID: %w", err)
```

Comment:
Not for just this, but in general, should we use Error sentinels? I made that update in some places a while ago, but we don't do it consistently everywhere. But we could do it wherever we introduce new errors going forward. I think we have an errors.go where we define the current sentinels.